added makefile

rcnijzink opened the pull request:
Addition of makefile as first trial

rcnijzink committed:
added makefile

rcnijzink committed:
removal of automatic restarts, options com_command redefined

rcnijzink committed:
vom_command changed in namelist

rcnijzink committed:
no return after sce initialization, sce continues

rcnijzink committed:
removal close statements in sce

rcnijzink committed:
added make check in makefile

rcnijzink committed:
added test directory

rcnijzink committed:
added sample.f90 to run vom randomly

rcnijzink committed:
added sample.f90 to run vom randomly

rcnijzink committed:
added .travis.yml

rcnijzink committed:
moved .travis file to right dir

rcnijzink committed:
travis file adjusted

rcnijzink committed:
travis file added install build essential

rcnijzink committed:
travis file adjusted

rcnijzink committed:
travis file adjusted

rcnijzink committed:
travis file adjusted

rcnijzink committed:
travis file adjusted

rcnijzink committed:
travis file moved and adjusted

rcnijzink committed:
deleted file

rcnijzink committed:
travis updated

rcnijzink committed:
travis updated

rcnijzink committed:
travis updated with make check

rcnijzink committed:
travis updated with changedir

rcnijzink committed:
travis typo path corrected

rcnijzink committed:
travis typo added extra make

rcnijzink committed:
added parameter i_iter in namelist

rcnijzink committed:
added filename and number in modules

rcnijzink committed:
added output in sample.f90

rcnijzink committed:
optmode replaced by vom_command for output

rcnijzink committed:
output added for random run

rcnijzink committed:
added extra outputfiles for random runs

schymans commented on VOM_Fortran/VOM-code/transpmodel.f90:
```diff
@@ -149,13 +150,19 @@ subroutine transpmodel(invar, dim_invar, tp_netass, option1)
+       if (option1 .eq. 5) then
+           if(nday .eq. 1) allocate( output_mat (21, c_testday ) )
+        call transpmodel_daily_step(tp_netass, option1, output_mat)
+       else
```
I don't see a difference between option1 .eq. 5 and "else". It seems that the following actions are identical. Could you verify and simplify or document?

rcnijzink replied:
Yes, I changed this already, this was just a temporary solution. The "option1" is now the trigger in the transpmodel_daily_step module.

schymans commented on VOM_Fortran/VOM-code/transpmodel.f90:
```diff
@@ -1528,6 +1540,90 @@ end subroutine vom_write_dayyear
 !+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 !+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 
+     subroutine vom_save_dayyear (output)
```
Could you add a short comment what this subroutine does?

rcnijzink replied:
I added a brief description

schymans commented on VOM_Fortran/VOM-code/transpmodel.f90:
```diff
@@ -1528,6 +1540,90 @@ end subroutine vom_write_dayyear
+!     &    etmg_y, assg_y, rlg_y, rrg_y, cpccg_y, tcg_y,                &
+!     &    etmt_y, asst_y, rlt_y, rrt_y, cpcct_y, tct_y
+!      endif
+
```
If the commented out code is not needed any more, it could be removed.

rcnijzink replied:
Removed commented code

schymans reviewed (changes requested):
Great work! Could you summarize briefly what this PR does, for future reference? I also added some comments to the code.

rcnijzink committed:
adjustments based on pull request

rcnijzink committed:
added parameter file as output

rcnijzink committed:
added input output options

schymans commented:
Do you know why Travis failed on this? Could you fix it? Thanks!

rcnijzink commented:
I added a variable "inputpath", which gfortran on travis didn't like in a stop statement, for a reason I really don't know (it was used without problems in the rest of the same subroutine). I adjusted the stop-statements as a workaround. It should run without problems now.

rcnijzink commented:
I close this pull-request, as it is a pull on the master of my fork. I'll make branches following a more correct git-workflow.

schymans commented:
OK, thanks!
